src/libraries/Pool.sol

```diff
@@ -265,13 +265,20 @@ library Pool {
     }
 
     struct SwapParams {
+        int256 amountSpecified;
```

Comment:
does this ordering change matter?

Reply:
we removed the cached `exactInput = params.amountSpecified < 0`, and are now using `params.amountSpecified < 0` in place of the cached variable

making it the first element in the struct avoids the need of offset calculations